dotCMS/src/main/java/com/dotcms/security/secret/ServiceDescriptor.java

```diff
@@ -0,0 +1,73 @@
+package com.dotcms.security.secret;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class ServiceDescriptor {
+
+    private String serviceKey;
+
+    private String name;
+
+    private String description;
+
+    private String iconUrl;
+
+    private Map<String,Param> params;
+
+    public ServiceDescriptor(final String serviceKey, final String name, final String description, final String iconUrl) {
+        this.serviceKey = serviceKey;
+        this.name = name;
+        this.description = description;
+        this.iconUrl = iconUrl;
+    }
+
+    public ServiceDescriptor() {
+    }
+
+    public String getServiceKey() {
+        return serviceKey;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String getIconUrl() {
+        return iconUrl;
+    }
+
+    public Map<String, Param> getParams() {
+        if(null == params){
+            params = new HashMap<>();
+        }
+        return params;
+    }
+
+    public void addParam(final String name, final String value, final boolean hidden,
+            final Type type, final String label, final String hint) {
+        getParams().put(name, Param.newParam(value, hidden, type, label, hint));
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'o' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)